### PR TITLE
desync: 0.4.0 -> 0.9.0

### PR DESCRIPTION
**Pull request by charles-dyfis-net**
###### Motivation for this change

The desync package has fallen far behind upstream, and consequently missed out on a great deal of feature and bugfix development.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
  46,885,312 -> 51,176,648 -- an increase, but not an out-of-range one given the dependency chain updates supporting new features.
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by doronbehar — CHANGES_REQUESTED** [review]
Upstream uses go modules now. Could you please also update the derivation to use `buildGoModule`? And remove `deps.nix`.

**Comment by charles-dyfis-net** [comment]
@doronbehar done, thank you -- I hadn't been aware of `buildGoModule`.

That said, with the cutover, the tests are no longer able to succeed -- it looks like the data they need at runtime (stored in the `testdata` and `cmd/desync/testdata` subdirectories of the desync source tree) is unavailable. I don't see any hints in the `buildGoModule` documentation on how to address that; would you have guidance?

File: pkgs/applications/networking/sync/desync/default.nix [hunk 1]
```diff
@@ -1,20 +1,21 @@
-{ stdenv, buildGoPackage, fetchFromGitHub }:
+{ stdenv, buildGoModule, fetchFromGitHub }:
 
-buildGoPackage rec {
+buildGoModule rec {
   pname = "desync";
-  version = "0.4.0";
+  version = "0.9.0";
   rev = "v${version}";
```
Comment (doronbehar) [inline pkgs/applications/networking/sync/desync/default.nix:6]: This has no use here IMO. Could you put it as is in the arguments to `fetchFromGitHub`?

**Review by doronbehar — CHANGES_REQUESTED** [review]
If you could please squash the commits - migrating to buildGoModule is part of the update.

File: pkgs/applications/networking/sync/desync/default.nix (hunk 1 above)
Comment (charles-dyfis-net) [reply pkgs/applications/networking/sync/desync/default.nix]: Done, as of 4f8057ec837d0c9a79930c21f1673ebc98074d67

**Review by doronbehar — APPROVED** [review]

**Comment by doronbehar** [comment]
> I don't see any hints in the `buildGoModule` documentation on how to address that; would you have guidance?

Me too :confused: . Tried to debug this a bit locally and failed to see why.

**Pull request merged**